content/developer/howtos/upgrade/upgrade_util.rst

```diff
@@ -0,0 +1,283 @@
+====================
+Upgrade Util package
+====================
+
+The `Upgrade Util package <https://github.com/odoo/upgrade-util/>`_ is a library that contains
+helper functions to facilitate the writing of upgrade scripts. This package, used by Odoo for the
+upgrade scripts of standard modules, provides reliability and helps speed up the upgrade process:
+
+- The helper functions make sure the data is consitent in the database.
+- It takes care of indirect references of the updated records.
+- Allows calling functions and avoid writing code, saving time and reducing development risks.
+- Helpers allow to focus on what is important for the upgrade and not think of details.
+
+Installation
+============
+
+Clone the `Upgrade Util repository <https://github.com/odoo/upgrade-util/>`_ locally and start
+``odoo`` with the ``src`` directory prepended to the ``--upgrade-path`` option.
+
+.. code-block:: console
+
+   $ ./odoo-bin --upgrade-path=/path/to/upgrade-util/src,/path/to/other/upgrade/script/directory [...]
+
+On platforms where you do not manage Odoo yourself, you can install this package via `pip`:
+
+.. code-block:: console
+
+   $ python3 -m pip install git+https://github.com/odoo/upgrade-util@master
+
+On `Odoo.sh <https://www.odoo.sh/>`_ it is recommended to add it to the :file:`requirements.txt` of
+the custom repository. For this, add the following line inside the file::
+
+   odoo_upgrade @ git+https://github.com/odoo/upgrade-util@master
+
+Using the Util package
+======================
+
+Once installed, the following packages are available for the upgrade scripts:
+
+- :mod:`odoo.upgrade.util`: the helper themself.
+- :mod:`odoo.upgrade.testing`: base TestCase classes.
+
+To use it in upgrade scripts, simply import it:
+
+.. code-block:: python
+
+   from odoo.upgrade import util
+
+
+   def migrate(cr, version):
+      # Rest of the script
+
+Now, the helper functions are available to be called through ``util``.
+
+Util functions
+==============
+
+The util package provides many useful functions to ease the upgrade process. Here, we describe some
+of the most useful ones. Refer to the `util folder
+<https://github.com/odoo/upgrade-util/tree/master/src/util>`_ for the comprehensive declaration of
+helper functions.
+
+.. note::
+
+   All util functions receive :attr:`cr` as a parameter. This refers to the database cursor. Use the
+   one received as a parameter in the :doc:`upgrade_scripts`.
+
+Fields
+------
+
+.. `[source] <https://github.com/odoo/upgrade-util/blob/master/src/util/fields.py#L91>`_
+.. method:: remove_field(cr, model, fieldname[, cascade=False][, drop_column=True][, skip_inherit=()])
+
+   Remove a field and its references from the database
+
+   :param str model: model name of the field to remove
+   :param str fieldname: name of the field to remove
+   :param bool cascade: if ``True``, removes the field's column and inheritance in ``CASCADE``
+      (default: ``False``)
```

Comment:
You can eventually use "Whether" to document booleans to make shorter sentences.
```suggestion
   :param bool cascade: Whether the field's column and inheritance are removed in ``CASCADE``
      (default: ``False``)
```
Same for other params and methods.